### ISSUE TITLE
auth InvalidScope

### ISSUE THREAD

**Issue by rquanx** [s-KaiNet/node-sp-auth#77]
Hi,

I use sp-rest-proxy in my project but get InvalidScope, it seems to be an authentication error

error message from  [node-sp-auth-troubleshoot](https://github.com/koltyakov/node-sp-auth-troubleshoot/issues/2)
```
Connecting to site: https://swirecocacola.sharepoint.cn/sites/scs

=== Error ===
 Error: <s:Fault>
  <s:Code>
    <s:Value>s:Sender</s:Value>
    <s:Subcode>
      <s:Value xmlns:a="http://docs.oasis-open.org/ws-sx/ws-trust/200512">a:InvalidScope</s:Value>
    </s:Subcode>
  </s:Code>
  <s:Reason>
    <s:Text xml:lang="en-US">ID3082: The request scope is not valid or is unsupported.</s:Text>
  </s:Reason>
</s:Fault>
    at config_1.request.post.then.xmlResponse (C:\Users\runquan.tan\Desktop\node-sp-auth-troubleshoot\node_modules\node-sp-auth\lib\src\utils\Adf
sHelper.js:31:23)
    at tryCatcher (C:\Users\runquan.tan\Desktop\node-sp-auth-troubleshoot\node_modules\bluebird\js\release\util.js:16:23)
    at Promise._settlePromiseFromHandler (C:\Users\runquan.tan\Desktop\node-sp-auth-troubleshoot\node_modules\bluebird\js\release\promise.js:547:
31)
    at Promise._settlePromise (C:\Users\runquan.tan\Desktop\node-sp-auth-troubleshoot\node_modules\bluebird\js\release\promise.js:604:18)        
    at Promise._settlePromise0 (C:\Users\runquan.tan\Desktop\node-sp-auth-troubleshoot\node_modules\bluebird\js\release\promise.js:649:10)       
    at Promise._settlePromises (C:\Users\runquan.tan\Desktop\node-sp-auth-troubleshoot\node_modules\bluebird\js\release\promise.js:729:18)       
    at _drainQueueStep (C:\Users\runquan.tan\Desktop\node-sp-auth-troubleshoot\node_modules\bluebird\js\release\async.js:93:12)
    at _drainQueue (C:\Users\runquan.tan\Desktop\node-sp-auth-troubleshoot\node_modules\bluebird\js\release\async.js:86:9)
    at Async._drainQueues (C:\Users\runquan.tan\Desktop\node-sp-auth-troubleshoot\node_modules\bluebird\js\release\async.js:102:5)
    at Immediate.Async.drainQueues [as _onImmediate] (C:\Users\runquan.tan\Desktop\node-sp-auth-troubleshoot\node_modules\bluebird\js\release\asy
nc.js:15:14)
    at runCallback (timers.js:705:18)
    at tryOnImmediate (timers.js:676:5)
    at processImmediate (timers.js:658:5)
```

Related issues 
- [JSOM contextinfo 400 InvalidScope](https://github.com/koltyakov/sp-rest-proxy/issues/100)
- [InvalidScope](https://github.com/koltyakov/node-sp-auth-troubleshoot/issues/2)

**Comment by s-KaiNet** [comment]
Hello, 

could I ask you to use Postman (or any alternative) to issue HTTP POST request to `https://login.microsoftonline.com/GetUserRealm.srf` with `login` form data as your login name?

The sample request I did:  

![image](https://user-images.githubusercontent.com/10658276/80650558-e4f41280-8a7c-11ea-9db1-22584a580fd3.png)

Please remove any sensitive data and submit the response here.

**Comment by rquanx** [comment]
Thanks!

It seems like your example
![P9 AHS8}@BQ`BX%0KHP((7L](https://user-images.githubusercontent.com/36622981/80667252-3b8b3c00-8ad1-11ea-9fb2-e3f6476cde5f.png)

**Comment by s-KaiNet** [comment]
Sorry, I've just noticed that your tenant is hosted in China. Then could you do the same, but this time for China cloud url - `https://login.chinacloudapi.cn/GetUserRealm.srf`

**Comment by rquanx** [comment]
Ok

![image](https://user-images.githubusercontent.com/36622981/80695355-60e76c80-8b08-11ea-88f0-c3ae26f76306.png)

**Comment by s-KaiNet** [comment]
Thank you, I have some ideas, later today I will create a test package for you to test if it works.

**Comment by s-KaiNet** [comment]
Please create a new folder and install a beta version of the library:

```bash
npm install node-sp-auth@beta
```

Then create a file `index.js` and add below code: 
```js
var spauth = require('node-sp-auth');

spauth.getAuth("<your sharepoint site url>", {
	username: "<username>",
	password: "<password>"
})
  .then(options => {
	  	 console.log(options.headers);
  });
```

Open a command line in the folder and run `node index.js`. Does it print valid headers or again throws an error?

**Comment by rquanx** [comment]
Yeah, it print valid headers

![image](https://user-images.githubusercontent.com/36622981/80748025-e0039180-8b56-11ea-892f-1d08b6bcd7db.png)

**Comment by s-KaiNet** [comment]
That's good news. I published `2.5.8`, which includes this fix.
